rskj-core/src/main/java/co/rsk/trie/Trie.java

```diff
@@ -106,7 +106,7 @@ private Trie(TrieStore store, TrieKeySlice sharedPath, byte[] value, boolean isS
 
     // full constructor
     private Trie(TrieKeySlice sharedPath, byte[] value, NodeReference left, NodeReference right, TrieStore store, boolean isSecure) {
-        this.value = value;
+        this.value = cloneArray(value);
```

Comment:
Maybe implement a private constructor that copies the reference (not the content) so that insertion/deletion operations keep being log2(N) and not dependent on the size of the value (e.g. a long bytecode).